File: problem/XOR_Multiplication_Queries_I/xorAfterQueries.js
/**
 * @param {number[]} nums
 * @param {number[][]} queries
 * @return {number}
 */
var xorAfterQueries = function(nums, queries) {
    const mod = 1000000007n;
    
    for (const [l, r, k, v] of queries) {
        const val = BigInt(v);
        for (let idx = l; idx <= r; idx += k) {
            nums[idx] = Number((BigInt(nums[idx]) * val) % mod);
        }
    }
    
    let ans = 0;
    for (const num of nums) {
        ans ^= num;
    }
    
    return ans;
};

File: problem/XOR_Multiplication_Queries_I/xorAfterQueries.py
class Solution:
    def xorAfterQueries(self, nums, queries):
        mod = 1000000007

        # Process each query
        for t in queries:
            l = t[0]
            r = t[1]
            k = t[2]
            v = t[3]

            idx = l

            # Apply operation at step k
            while idx <= r:
                temp = nums[idx]
                nums[idx] = (temp * v) % mod
                idx += k

        # Compute XOR of final array
        ans = 0
        for num in nums:
            ans ^= num

        return ans